### PR TITLE
fix/resolve_symlinks

### LINKED ISSUE #1009: name 'config' is not defined on second resubmission with input: all_directions: "link"
```
oifs:
    file_movements:
        forcing:
            all_directions: "link"
        input:
            all_directions: "link"
```
reproducibly results in:
```

::: Copying files

::: copying file category: ICMGG_INIT
- source: /work/ab0246/a270092/runtime/awicm3-v3.1.3/link_test/input/oifs/ICMGGawi3INIT
- target: /work/ab0246/a270092/runtime/awicm3-v3.1.3/link_test/run_18500501-18500630/input/oifs/ICMGGawi3INIT
Traceback (most recent call last):
  File "/home/a/a270092/.local/bin/esm_runscripts", line 33, in <module>
    sys.exit(load_entry_point('esm-tools', 'console_scripts', 'esm_runscripts')())
  File "/home/a/a270092/esm_tools/src/esm_runscripts/cli.py", line 278, in main
    Setup()
  File "/home/a/a270092/esm_tools/src/esm_runscripts/sim_objects.py", line 90, in __call__
    resubmit.maybe_resubmit(self.config)
  File "/home/a/a270092/esm_tools/src/esm_runscripts/resubmit.py", line 148, in maybe_resubmit
    nextrun = resubmit_recursively(config, jobtype=jobtype)
  File "/home/a/a270092/esm_tools/src/esm_runscripts/resubmit.py", line 189, in resubmit_recursively
    resubmit_SimulationSetup(config, cluster)
  File "/home/a/a270092/esm_tools/src/esm_runscripts/resubmit.py", line 66, in resubmit_SimulationSetup
    config["general"]["experiment_over"] = cluster_obj(kill_after_submit=False)
  File "/home/a/a270092/esm_tools/src/esm_runscripts/sim_objects.py", line 90, in __call__
    resubmit.maybe_resubmit(self.config)
  File "/home/a/a270092/esm_tools/src/esm_runscripts/resubmit.py", line 164, in maybe_resubmit
    nextrun = resubmit_recursively(
  File "/home/a/a270092/esm_tools/src/esm_runscripts/resubmit.py", line 189, in resubmit_recursively
    resubmit_SimulationSetup(config, cluster)
  File "/home/a/a270092/esm_tools/src/esm_runscripts/resubmit.py", line 66, in resubmit_SimulationSetup
    config["general"]["experiment_over"] = cluster_obj(kill_after_submit=False)
  File "/home/a/a270092/esm_tools/src/esm_runscripts/sim_objects.py", line 69, in __call__
    self.prepcompute()
  File "/home/a/a270092/esm_tools/src/esm_runscripts/sim_objects.py", line 161, in prepcompute
    self.config = prepcompute.run_job(self.config)
  File "/home/a/a270092/esm_tools/src/esm_runscripts/prepcompute.py", line 32, in run_job
    config = evaluate(config, "prepcompute", "prepcompute_recipe")
  File "/home/a/a270092/esm_tools/src/esm_runscripts/helpers.py", line 69, in evaluate
    config = esm_plugin_manager.work_through_recipe(
  File "/home/a/a270092/esm_tools/src/esm_plugin_manager/esm_plugin_manager.py", line 141, in work_through_recipe
    config = getattr(submodule, workitem)(config)
  File "/home/a/a270092/esm_tools/src/esm_runscripts/prepcompute.py", line 240, in copy_files_to_thisrun
    config = copy_files(
  File "/home/a/a270092/esm_tools/src/esm_runscripts/filelists.py", line 959, in copy_files
    file_source = resolve_symlinks(file_source)
  File "/home/a/a270092/esm_tools/src/esm_runscripts/filelists.py", line 902, in resolve_symlinks
    if config["general"]["verbose"]:
NameError: name 'config' is not defined
```
during the tidy phase of the second leg, as the work folder for the leg 3 is being made. **I'm currently making a second test if using sub yearly restarts is another required factor.** I think it might be, because we have been using the above options for AWI-ESM1 and for AWI-CM3 TCO319-DART on aleph. Only once I set the restart freq to every 2 months for TCO1279-DART did I start to see this issue.

My test is TCO95-CORE2 with the same file movements but yearly restarts. I already did the bi-monthly restarts with normal / copy file movement and that did not crash.

### PR DESCRIPTION
**Pull request by mandresm**
Closes #1009

### CODE REVIEW

File: src/esm_runscripts/filelists.py [hunk 1]
```diff
@@ -892,7 +892,7 @@ def check_for_unknown_files(config):
     return config
 
 
-def resolve_symlinks(file_source):
+def resolve_symlinks(config, file_source):
```
Comment (pgierz) [inline src/esm_runscripts/filelists.py:895]: How did this ever work in the first place? config is never defined in the local scope???

File: src/esm_runscripts/filelists.py [hunk 2]
```diff
@@ -905,7 +905,7 @@ def resolve_symlinks(file_source):
             return file_source
 
         # recursively find the file that the link is pointing to
-        return resolve_symlinks(points_to)
+        return resolve_symlinks(config, points_to)
```
Comment (pgierz) [inline src/esm_runscripts/filelists.py:908]: Same comment as above

**Review by pgierz — APPROVED** [review]
Thanks Miguel. 👍🏻 I am still confused a little bit about how this ever worked in the first place. I guess `file_source` was always directly the same as `points_to`, but this was a very sloppy mistake ☹️

**Comment by mandresm** [comment]
#bump

**Pull request merged**